### PR TITLE
granite: update to 5.1.0.

### PR DESCRIPTION
**Pull request by TeddyDD**
I haven't updated library yet, so I'm not sure if I done everything correctly. I removed patch since it builds without it.

- [x]  `io.elementary.videos-2.6.1_1` fails to build due to type error (uint/int mismatch)
- [x] `pantheon-screenshot-0.1.5_1` fails due to checksum mismatch 

Should I fix those first? Rest of dependent packages builds without errors.

### CODE REVIEW

**Comment by CameronNemo** [comment]
https://github.com/elementary/videos/commit/59dc6d61b0ad4a7e91cdda076169cd328f58a0c8

File: srcpkgs/granite/template [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-create_wrksrc=yes
 build_style=cmake
 hostmakedepends="pkg-config gobject-introspection vala-devel libgee08-devel"
 makedepends="gtk+3-devel libgee08-devel"
 depends="hicolor-icon-theme"
 short_desc="GTK+3 extension providing commonly-used widgets"
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
-homepage="https://launchpad.net/granite"
+homepage="https://github.com/elementary/granite"
 license="LGPL-3"
```
Comment (maxice8) [inline srcpkgs/granite/template]: please use spdx short-identifier

**Comment by maxice8** [comment]
> I tried to build all dependent packages.

bump revision= on packages that depend on it

`xrevbump 'rebuild against libgranite.so.5' <pkg> ...`

**Comment by TeddyDD** [comment]
ummm, should I squash that?

**Comment by maxice8** [comment]
No, each revbump should be separate

**Comment by maxice8** [comment]
regenerate the checksum for pantheon-screenshot

**Comment by maxice8** [comment]
Please fix wrksrc= for pantheon screenshot.

Put it in the same commit as the revbump, also put the checksum regeneration in the same commit

**Comment by TeddyDD** [comment]
@CameronNemo  looks like commit you linked is already included in current version of 
videos, and It should build. Valac should define `VALA_0_42` but it doesn't). Dunno why. I can't figure out how to pass this flag to compiler.

https://wiki.gnome.org/Projects/Vala/Manual/Preprocessor

**Comment by CameronNemo** [comment]
Building it with fresh granite works locally. I have a patch that unconditionally makes the change, without preprocessor tests. If you can't figure out how to make it build with the current patch applied I will submit a PR.

**Comment by TeddyDD** [comment]
@CameronNemo Yup, works with your patch, thank you :)
 I thought it supposed to build as it, but oh well.

**Comment by CameronNemo** [comment]
Do you want me to make a separate PR or will you include the patch here? I am fine with either.

**Comment by TeddyDD** [comment]
wtf, I see your patch locally, it looks like committed, but it's not here. Hooray, I broken git. I'll add your patch here when I fix my local repo...

**Comment by CameronNemo** [comment]
This looks good to merge from my end!

**Pull request merged**